### PR TITLE
AP_ADSB: avoid using GPS and RTC libraries in backend

### PR DESCRIPTION
**Pull request by peterbarker**
instead, add fields to canonical AP_ADSB location "_my_loc" to hold all of the information backends might want.  This will allow consistent presentation of data regardless of backend type, and for the sources of the information to change more easily.

This work sponsored by SpektreWorks

### CODE REVIEW

File: libraries/AP_ADSB/AP_ADSB.h [hunk 1]
```diff
@@ -85,6 +86,74 @@ class AP_ADSB {
...
+            }
+            hacc = horizontal_pos_accuracy;
+            return true;
+        }
+        bool vertical_accuracy(float &vacc) const {
+            if (!vertical_pos_accuracy_is_valid) {
+                return false;
+            }
+            vacc = vertical_pos_accuracy;
+            return true;;
```
Comment (tridge) [inline libraries/AP_ADSB/AP_ADSB.h:142]: more ; are good ; ?

File: libraries/AP_ADSB/AP_ADSB.h [hunk 2]
```diff
@@ -85,6 +86,74 @@ class AP_ADSB {
...
+        }
+        uint8_t num_sats() const { return satellites; }
+
+        // methods to make us look like the AP::ahrs() singleton:
+        Vector2f groundspeed_vector() const { return vel_ned.xy(); }
```
Comment (tridge) [inline libraries/AP_ADSB/AP_ADSB.h:147]: const ref?

File: libraries/AP_ADSB/AP_ADSB.h [hunk 3]
```diff
@@ -85,6 +86,74 @@ class AP_ADSB {
...
+        bool horizontal_vel_accuracy_is_valid;
+
+        Vector3f vel_ned;
+
+        float vertRateD;  // m/s down
+        bool vertRateD_is_valid;
+
+        // methods to make us look much like the AP::gps() singleton:
+        AP_GPS_FixType status() const { return fix_type; }
+        Vector3f velocity() const {
```
Comment (tridge) [inline libraries/AP_ADSB/AP_ADSB.h:119]: const ref?

File: libraries/AP_ADSB/AP_ADSB.h [hunk 4]
```diff
@@ -85,6 +86,74 @@ class AP_ADSB {
...
+            return vel_ned;
+        }
+        uint64_t time_epoch_usec() const { return epoch_us; }
+        bool speed_accuracy(float &sacc) const {
```
Comment (tridge) [inline libraries/AP_ADSB/AP_ADSB.h:123]: better in cpp?

File: libraries/AP_ADSB/AP_ADSB.h [hunk 5]
```diff
@@ -85,6 +86,74 @@ class AP_ADSB {
...
+            if (!horizontal_vel_accuracy_is_valid) {
+                return false;
+            }
+            sacc = horizontal_vel_accuracy;
+            return true;
+        }
+        bool horizontal_accuracy(float &hacc) const {
```
Comment (tridge) [inline libraries/AP_ADSB/AP_ADSB.h:130]: in cpp

**Review by tridge — CHANGES_REQUESTED** [review]
ok with concept

**Review by peterbarker — COMMENTED** [review]
Fixed review comments

**Review by tridge — APPROVED** [review]

**Pull request merged**